WooCommerce/Classes/Extensions/UINavigationController+Woo.swift

```diff
@@ -75,3 +75,34 @@ extension UINavigationController: UINavigationBarDelegate {
         return false
     }
 }
+
+// MARK: - Handle the swipe back gesture
+protocol NavigationSwipeBackHandler {
+
+    /// Should block the 'SwipeBack' gesture
+    ///
+    /// - Returns: true - don't block，false - block
+    func shouldPopOnSwipeBack() -> Bool
+}
+
+extension UIViewController: NavigationSwipeBackHandler {
+    //Do not block the "Swipe back" gesture by default, otherwise, override this function in the specified viewcontroller
+    @objc func shouldPopOnSwipeBack() -> Bool {
+        return true
+    }
+}
+
+extension UIViewController: UIGestureRecognizerDelegate {
+
+    public func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldBeRequiredToFailBy otherGestureRecognizer: UIGestureRecognizer) -> Bool {
+        if gestureRecognizer.isEqual(navigationController?.interactivePopGestureRecognizer) {
+            return shouldPopOnSwipeBack()
+        }
+
+        return false
+    }
+
+    func handleSwipeBackGesture() {
+        navigationController?.interactivePopGestureRecognizer?.delegate = self
```

Comment:
I'm not 100% about embedding the assigning of the delegate in a method like this, but it's better understandable than declaring the delegate in every view controller. Do you have any suggestion about it?